File: src/main/java/mc/fbturrets/blocks/ITurretGun.java
package mc.fbturrets.blocks;

import net.minecraft.particle.DefaultParticleType;
import net.minecraft.util.math.Box;

public interface ITurretGun {
    int getAimTime();
    double getRotSpeed();
    int getDamage();
    DefaultParticleType getEffect();
    Box getTargetBox();
}

File: src/main/kotlin/mc/fbturrets/blocks/ITurretGun.kt
package mc.fbturrets.blocks

import net.minecraft.particle.DefaultParticleType
import net.minecraft.util.Identifier
import net.minecraft.util.math.Box

interface ITurretGun {
    val aimTime: Int
    val rotSpeed: Double
    val damage: Int
    val effect: DefaultParticleType
    val targetBox: Box
    val iD: Identifier
}